clippy_lints/src/types.rs

```diff
@@ -1215,12 +1215,14 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Casts {
                 if from_align < to_align;
                 // with c_void, we inherently need to trust the user
                 if !is_c_void(cx, from_ptr_ty.ty);
+                // when casting from a ZST, we don't know enough to properly lint
+                if !(Some(true) == cx.layout_of(from_ptr_ty.ty).ok().map(|a| a.is_zst()));
                 then {
                     span_lint(
                         cx,
                         CAST_PTR_ALIGNMENT,
                         expr.span,
-                        &format!("casting from `{}` to a more-strictly-aligned pointer (`{}`)", cast_from, cast_to)
+                        &format!("casting from `{}` to a more-strictly-aligned pointer (`{}`) ({} < {})", cast_from, cast_to, from_align.bytes(), to_align.bytes())
```

Comment:
Could you add the word "bytes" so the user knows the unit of these numbers when reading the message?

```suggestion
                        &format!("casting from `{}` to a more-strictly-aligned pointer (`{}`) ({} < {} bytes)", cast_from, cast_to, from_align.bytes(), to_align.bytes())
```